geomesa-gs-catalog-listener/src/main/scala/org/geomesa/gs/catalog/GeoMesaCatalogListener.scala

```diff
@@ -79,13 +83,34 @@ class GeoMesaCatalogListener extends CatalogListener with InitializingBean with
   }
 
   override def handleAddEvent(event: CatalogAddEvent): Unit = {
-    logger.debug(s"GeoMesa Catalog Listener received add event: $event")
+    logger.debug(s"GeoMesa Catalog Listener received add event: $event of type ${event.getSource.getClass}")
     event.getSource match {
       case fti: FeatureTypeInfo => addFeatureTypeInfo(fti)
-      case _ => // not a new layer; no action necessary.
+      case dti: DataStoreInfo => registerLayers(dti)
+      case wsi: WorkspaceInfo => registerDataStore(wsi)
+      case _ => event.getSource.getClass  // not a new layer; no action necessary.
     }
   }
 
+  def registerDataStore(wsi: WorkspaceInfo): Unit = {
+    val base = System.getProperty("GEOMESA_FSDS_BASE_DIRECTORY")
+    val directory = wsi.getName
+
+    val dsi = catalog.getFactory.createDataStore()
+    dsi.setWorkspace(wsi)
+    dsi.setEnabled(true)
+    dsi.setName(directory)
+
+    val factory = new FileSystemDataStoreFactory
```

Comment:
what happens if you're not using the FSDS? is this going to break all other datastores?